crates/egui/src/widgets/text_edit/builder.rs

```diff
@@ -670,17 +678,12 @@ impl<'t> TextEdit<'t> {
                 let galley = if multiline {
                     hint_text.into_galley(
                         ui,
-                        Some(TextWrapMode::Wrap),
+                        Some(wrap_mode),
                         desired_inner_size.x,
                         hint_text_font_id,
                     )
                 } else {
-                    hint_text.into_galley(
-                        ui,
-                        Some(TextWrapMode::Extend),
-                        f32::INFINITY,
-                        hint_text_font_id,
-                    )
+                    hint_text.into_galley(ui, Some(wrap_mode), f32::INFINITY, hint_text_font_id)
```

Comment:
I think single-line text-edits should always use `Extend`, otherwise they are not single-line